maven-plugins/bnd-generate-maven-plugin/src/main/java/aQute/bnd/maven/generate/plugin/GenerateMojo.java

```diff
@@ -1,222 +1,17 @@
-import org.apache.maven.settings.Settings;
-import org.eclipse.aether.RepositorySystem;
-import org.eclipse.aether.RepositorySystemSession;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-@Mojo(name = "bnd-generate", defaultPhase = LifecyclePhase.GENERATE_SOURCES, requiresDependencyResolution = ResolutionScope.NONE, threadSafe = true)
-public class GenerateMojo extends AbstractMojo {
-
-	protected final Logger			logger	= LoggerFactory.getLogger(getClass());
-
-	@Parameter(defaultValue = "${project}", readonly = true, required = true)
-	private MavenProject										project;
-
-	@Parameter(defaultValue = "${settings}", readonly = true)
-	private Settings											settings;
-
-	@Parameter(defaultValue = "${mojoExecution}", readonly = true)
-	MojoExecution												mojoExecution;
-
-	@Parameter(defaultValue = "${repositorySystemSession}", readonly = true, required = true)
-	private RepositorySystemSession								repositorySession;
-
-	@Parameter(defaultValue = "${project.build.directory}", readonly = true)
-	private File												targetDir;
 
-	/**
-	 * Skip this goal.
-	 */
-	@Parameter(property = "bnd.generate.skip", defaultValue = "false")
-	boolean														skip;
-
-	/**
-	 * The Mojo by itself does nothing. Dependent on its configuration, it will
-	 * try to find a suitable generator in one of the dependencies in this list.
-	 */
-	@Parameter(property = "externalPlugins", required = false)
-	List<Dependency>				externalPlugins;
-
-	/**
-	 * Allows multiple steps
-	 */
-	@Parameter(property = "steps", required = false)
-	List<Step>						steps;
-
-	@Parameter(defaultValue = "${session}", readonly = true)
-	private MavenSession										session;
-
-	@Component
-	private RepositorySystem									system;
-
-	/**
-	 * File path to a bnd file containing bnd instructions for this project.
-	 * Defaults to {@code bnd.bnd}. The file path can be an absolute or relative
-	 * to the project directory.
-	 * <p>
-	 * The bnd instructions for this project are merged with the bnd
-	 * instructions, if any, for the parent project.
-	 */
-	@Parameter(defaultValue = Project.BNDFILE)
-	// This is not used and is for doc only; see {@link
-	// BndConfiguration#loadProperties(Processor)}
-	@SuppressWarnings("unused")
-	String							bndfile;
-
-	/**
-	 * Bnd instructions for this project specified directly in the pom file.
-	 * This is generally be done using a {@code <![CDATA[]]>} section. If the
-	 * projects has a {@link #bndfile}, then this configuration element
-	 * is ignored.
-	 * <p>
-	 * The bnd instructions for this project are merged with the bnd
-	 * instructions, if any, for the parent project.
-	 */
-	@Parameter
-	// This is not used and is for doc only; See {@link
-	// BndConfiguration#loadProperties(Processor)}
-	@SuppressWarnings("unused")
-	String							bnd;
+@Mojo(name = "generate", defaultPhase = LifecyclePhase.GENERATE_SOURCES, requiresDependencyResolution = ResolutionScope.RUNTIME_PLUS_SYSTEM, threadSafe = true)
```

Comment:
This should be `ResolutionScope.COMPILE` (compile + system + provided dependencies). Runtime dependencies should not be included since the generated code will be compiled with compile scope dependencies.

Reply:
Done